packages/billing/lib/batcher.ts

```diff
@@ -0,0 +1,125 @@
+import type { Result } from '@nangohq/utils';
+import { Err, Ok } from '@nangohq/utils';
+
+interface Item<T> {
+    item: T;
+    retries: number;
+}
+
+export class Batcher<T> {
+    private readonly process: (events: T[]) => Promise<void>;
+    private readonly maxBatchSize: number;
+    private readonly flushInterval: number;
+    private queue: Item<T>[];
+    private readonly maxQueueSize: number;
+    private readonly maxProcessingRetry: number;
```

Comment:
in order to avoid `poisoned` events that would be retried for ever. Default value is set to 3 retries